astropy/coordinates/jparser.py

```diff
@@ -0,0 +1,81 @@
+import re
+
+import numpy as np
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+
+def _sexagesimal(g):
+    # convert matched regex groups to sexigesimal array
+    sign, h, m, s, frac = g
+    sign = [1, -1][sign == '-']
+    s = '.'.join((s, frac))
+    return sign * np.array([h, m, s], float)
+
+
+class Jparser(object):
```

Comment:
From the code here I don't see any reason to make this a class.  You can just remove the class definition and turn the object methods into module functions, and don't make the static singleton `parser` object at the end.

Given the way this implementation relies on a module-level instance of `Jparser()`, there is not even opportunity to subclass from this in a useful way.  It might be that one could generalize this concept to introduce a subclassable parser that users could use.  Then you likely want a metaclass and registry.